routing/pathfind.go

```diff
@@ -196,6 +191,23 @@ type Route struct {
 	prevHopMap map[Vertex]*Hop
 }
 
+// HopFee returns the fee charged by the route hop indicated by hopIndex.
+func (r *Route) HopFee(hopIndex int) lnwire.MilliSatoshi {
+	if hopIndex < 0 || hopIndex >= len(r.Hops) {
+		panic(fmt.Errorf("invalid hop index"))
```

Comment:
I don't this this check is necessary, we can instead let it crash indexing into the Hops.